pyrtls.pyi

```diff
@@ -0,0 +1,407 @@
+    The server name must be either a DNS hostname or an IP address (only string forms are
+    currently accepted).
+    """
+
+    def __new__(cls, config: ClientConfig, name: str) -> ClientConnection: ...
+    def readable(self) -> bool:
+        """
+        Returns `true` if the caller should call `read_tls()` as soon as possible.
+
+        If there is pending plaintext data to read, this returns `false`. If your application
+        respects this mechanism, only one full TLS message will be buffered by pyrtls.
+        """
+
+    def read_tls(self, buf: list[int]) -> int:
+        """
+        Read TLS content from `buf` into the internal buffer. Return the number of bytes read, or
+        `0` once a `close_notify` alert has been received. No additional data is read in this
+        state.
+
+        Due to internal buffering, `buf` may contain TLS messages in arbitrary-sized chunks (like
+        a socket or pipe might).
+
+        You should call `process_new_packets()` each time a call to this function succeeds in
+        order to empty the incoming TLS data buffer.
+
+        Exceptions may be raised to signal backpressure:
+
+        * In order to empty the incoming TLS data buffer, you should call `process_new_packets()`
+          each time a call to this function succeeds.
+        * In order to empty the incoming plaintext data buffer, you should call `read_into()`
+          after the call to `process_new_packets()`.
+
+        You should call `process_new_packets()` each time a call to this function succeeds in
+        order to empty the incoming TLS data buffer
+
+        Mirrors the `RawIO.write()` interface.
+        """
+
+    def process_new_packets(self) -> IoState:
+        """
+        Processes any new packets read by a previous call to `read_tls()`.
+
+        Errors from this function relate to TLS protocol errors, and are fatal to the connection.
+        Future calls after an error will do no new work and will return the same error. After an
+        error is returned from `process_new_packets()`, you should not call `read_tls()` anymore
+        (it will fill up buffers to no purpose). However, you may call the other methods on the
+        connection, including `write()` and `write_tls_into()`. Most likely you will want to
+        call `write_tls_into()` to send any alerts queued by the error and then close the
+        underlying connection.
+
+        In case of success, yields an `IoState` object with sundry state about the connection.
+        """
+
+    def write(self, buf: list[int]) -> int:
+        """
+        Send the plaintext `buf` to the peer, encrypting and authenticating it. Once this
+        function succeeds you should call `write_tls_into()` which will output the
+        corresponding TLS records.
+
+        This function buffers plaintext sent before the TLS handshake completes, and sends it as
+        soon as it can.
+        """
+
+    def writeable(self) -> bool:
```

Comment:
I spelled it `writable()` (no "e"), which seems to be more common.

Reply:
Fixed in the next commit.